### DISCUSSION TITLE
Partial recipes

### DISCUSSION THREAD

**Discussion by Abrynos** [cooklang/spec#97] [Ideas]
Often times recipes have multiple parts. For example when making lasagna we have:
- Béchamelsauce
- Bolognese
- Preparing the baking dish and combining final ingredients

I'm proposing adding headlines similar to what markdown does to partition a recipe:

```
>> time required: 1.5 hours
>> course: dinner

# Bechamelsauce

Do whatever with @milk and @flour

Magic

Ready


# Bologenese

Cut @stuff

Cook stuff with @oil in #pan


# Baking

preheat oven

Prepare #baking dish{}

layer stuff in

top off with @cheese

bake
```

Since `#` is already used for cookware, that might not be the best option. Open for other suggestions.

**Comment by dubadub** [comment]
Yeah, maybe we should stir away from using `#`. 

Also, just for the sake of idea, another option could be to use alternative old-school syntax for headers like that 

```
>> time required: 1.5 hours
>> course: dinner

Bechamelsauce
============

Do whatever with @milk and @flour

Magic

Ready


Bologenese
=========

Cut @stuff

Cook stuff with @oil in #pan


Baking
=====

preheat oven

Prepare #baking dish{}

layer stuff in

top off with @cheese

bake
```

The only problem it works well with monospaced fonts...

**Comment by Zheoni** [comment]
My take on this was with `=` at the beginning of the line. To help with visually dividing the recipe in the `.cook`, more `=` are allowed (but not required) before and after the header text. Note that the number of `=` does NOT imply any nesting level.

```
>> time required: 1.5 hours
>> course: dinner

=== Bechamelsauce ===
-- how to prepare bechamel sauce ...

=== Bologenese
-- prepare bolognese

= Baking
-- combine everything and bake
```

This is how it's implemented in [`cooklang-rs`](https://github.com/cooklang/cooklang-rs) as an [extension](https://github.com/cooklang/cooklang-rs/blob/main/extensions.md), so if it gets into the spec we only have to make it enabled by default. I'm not trying to force this syntax, just giving context on where it comes from.

**Comment by dubadub** [comment]
Closing as it was added to the specification and implemented in the apps.
